src/hotspot/share/jvmci/jvmci.cpp

```diff
@@ -228,9 +229,17 @@ void JVMCI::vlog(int level, const char* format, va_list ap) {
 void JVMCI::vtrace(int level, const char* format, va_list ap) {
   if (JVMCITraceLevel >= level) {
     Thread* thread = Thread::current_or_null_safe();
-    if (thread != nullptr) {
+    if (thread != nullptr && thread->is_Java_thread()) {
       ResourceMark rm;
-      tty->print("JVMCITrace-%d[%s]:%*c", level, thread->name(), level, ' ');
+      JavaThreadState state = ((JavaThread*) thread)->thread_state();
```

Comment:
Please use `JavaThread::cast(thread)`

Reply:
Yes `JavaThread::cast(t)` is preferred. We did a lot of cleanup work ensuring we use `JavaThread` when always dealing with a `JavaThread`, and so reduce the places we need to cast. I'm a little surprised we still have some raw casts left as I thought we had cleaned them all up. I will check and file another cleanup RFE. Thanks.